src/test/regress/expected/update_gp.out

```diff
@@ -209,6 +208,418 @@ DROP TABLE keo2;
+                                             ->  Seq Scan on t1
+ Optimizer: Postgres query optimizer
+(12 rows)
+
+WITH CTE AS (DELETE FROM t1 RETURNING *)
+SELECT count(*) AS a FROM t_strewn JOIN cte USING (i);
+ a 
+---
+ 4
+(1 row)
+
+EXPLAIN (costs off)
+DELETE FROM t_strewn WHERE t_strewn.i = (SELECT t2.i FROM t2 WHERE t_strewn.i = t2.i);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Delete on t_strewn
+   ->  Seq Scan on t_strewn
+         Filter: (i = (SubPlan 1))
+         SubPlan 1  (slice0; segments: 3)
+           ->  Result
+                 Filter: (t_strewn.i = t2.i)
+                 ->  Materialize
+                       ->  Broadcast Motion 3:3  (slice1; segments: 3)
+                             ->  Seq Scan on t2
+ Optimizer: Postgres query optimizer
+(10 rows)
+
+DELETE FROM t_strewn WHERE t_strewn.i = (SELECT t2.i FROM t2 WHERE t_strewn.i = t2.i)
+RETURNING *;
+ i  
+----
+  3
+  9
+  6
+ 12
+(4 rows)
+
+EXPLAIN (costs off)
+UPDATE t_strewn SET i = t_strewn2.i
+FROM t_strewn2 WHERE t_strewn.i = t_strewn2.i;
+                               QUERY PLAN                               
+------------------------------------------------------------------------
+ Update on t_strewn
+   ->  Explicit Redistribute Motion 3:3  (slice3; segments: 3)
+         ->  Hash Join
+               Hash Cond: (t_strewn.i = t_strewn2.i)
+               ->  Redistribute Motion 3:3  (slice1; segments: 3)
+                     Hash Key: t_strewn.i
+                     ->  Seq Scan on t_strewn
+               ->  Hash
+                     ->  Redistribute Motion 3:3  (slice2; segments: 3)
+                           Hash Key: t_strewn2.i
+                           ->  Seq Scan on t_strewn2
+ Optimizer: Postgres query optimizer
+(12 rows)
+
+UPDATE t_strewn SET i = t_strewn2.i
+FROM t_strewn2 WHERE t_strewn.i = t_strewn2.i
+RETURNING *;
+ i  | i  
+----+----
+  5 |  5
+ 13 | 13
+ 15 | 15
+ 10 | 10
+ 11 | 11
+  2 |  2
+  7 |  7
+  8 |  8
+ 14 | 14
+  4 |  4
+ 16 | 16
+(11 rows)
+
+DROP TABLE t1;
+DROP TABLE t2;
+DROP TABLE t_strewn;
+DROP TABLE t_strewn2;
+-- Explicit Redistribute Motion should not be added if there are motions in Init
+-- Plans but not in the main plan. (test case not applicable to ORCA)
+CREATE TABLE t1 (a int, b int) DISTRIBUTED RANDOMLY;
+CREATE TABLE t2 (a int, b int) DISTRIBUTED RANDOMLY;
+INSERT INTO t1 SELECT
+  generate_series(1, 4) * 3, generate_series(1, 4);
+INSERT INTO t2 SELECT
+  generate_series(1, 32), generate_series(1, 32) * 3;
+ANALYZE t1;
+ANALYZE t2;
+EXPLAIN (costs off)
+WITH cte AS (
+    SELECT count(*) AS c FROM t1, t2 WHERE t1.b = t2.b
+) DELETE FROM t2 WHERE a = (SELECT * FROM cte);
+                                    QUERY PLAN                                     
+-----------------------------------------------------------------------------------
+ Delete on t2
+   InitPlan 1 (returns $0)  (slice3)
```

Comment:
There is no motion in InitPlan above t2. Motion is above t2_1

Reply:
It seems like scans on the same table in InitPlan behave the same way as without them: if there's another scan on the same table, the table will get relabeled (_1 appended to the name) and will use a different range table index. Initially, this test was added when we were still comparing relations' oids instead of range table indexes, so tests and the logic related to InitPlans is redundant now.